pkg/config/utils/metadata_as_tags.go

```diff
@@ -0,0 +1,173 @@
+}
+
+// GetNamespaceLabelsAsTags returns namespace labels as tags
+func (m *MetadataAsTags) GetNamespaceLabelsAsTags() map[string]string {
+	return m.labelsAsTags["namespaces"]
+}
+
+// GetNamespaceAnnotationsAsTags returns namespace annotations as tags
+func (m *MetadataAsTags) GetNamespaceAnnotationsAsTags() map[string]string {
+	return m.annotationsAsTags["namespaces"]
+}
+
+// GetResourcesLabelsAsTags returns a map from group resource to labels as tags
+func (m *MetadataAsTags) GetResourcesLabelsAsTags() map[string]map[string]string {
+	return m.labelsAsTags
+}
+
+// GetResourcesAnnotationsAsTags returns a map from group resource to annotations as tags
+func (m *MetadataAsTags) GetResourcesAnnotationsAsTags() map[string]map[string]string {
+	return m.annotationsAsTags
+}
+
+func (m *MetadataAsTags) mergeGenericResourcesLabelsAsTags(cfg pkgconfigmodel.Reader) {
+	resourcesToLabelsAsTags := retrieveDoubleMappingFromConfig(cfg, "kubernetes_resources_labels_as_tags")
+
+	for resource, labelsAsTags := range resourcesToLabelsAsTags {
+		// "pods.", "nodes.", "namespaces." are valid configurations, but they should be replaced here by "pods", "nodes" and "namespaces" respectively to ensure that they override the existing configurations for pods, nodes and namespaces
+		cleanResource := strings.TrimSuffix(resource, ".")
+		_, found := m.labelsAsTags[cleanResource]
+		if !found {
+			m.labelsAsTags[cleanResource] = map[string]string{}
+		}
+		// When a key in `labelsAsTags` exist in `m.labelsAsTags[cleanResource]`, the value in `m.labelsAsTags[cleanResource]` will be overwritten by the value associated with the key in `labelsAsTags`
+		// source: https://pkg.go.dev/maps#Copy
+		maps.Copy(m.labelsAsTags[cleanResource], labelsAsTags)
+	}
+}
+
+func (m *MetadataAsTags) mergeGenericResourcesAnnotationsAsTags(cfg pkgconfigmodel.Reader) {
+	resourcesToAnnotationsAsTags := retrieveDoubleMappingFromConfig(cfg, "kubernetes_resources_annotations_as_tags")
+
+	for resource, annotationsAsTags := range resourcesToAnnotationsAsTags {
+		// "pods.", "nodes.", "namespaces." are valid configurations, but they should be replaced here by "pods", "nodes" and "namespaces" respectively to ensure that they override the existing configurations for pods, nodes and namesapces
+		cleanResource := strings.TrimSuffix(resource, ".")
+		_, found := m.annotationsAsTags[cleanResource]
+		if !found {
+			m.annotationsAsTags[cleanResource] = map[string]string{}
+		}
+		// When a key in `annotationsAsTags` exist in `m.annotationsAsTags[cleanResource]`, the value in `m.annotationsAsTags[cleanResource]` will be overwritten by the value associated with the key in `annotationsAsTags`
+		// source: https://pkg.go.dev/maps#Copy
+		maps.Copy(m.annotationsAsTags[cleanResource], annotationsAsTags)
+	}
+}
+
+// GetMetadataAsTags returns a merged configuration of all labels and annotations as tags set by the user
+func GetMetadataAsTags(c pkgconfigmodel.Reader) MetadataAsTags {
```

Comment:
I thought a bit about having a global shared `MetadataAsTags` object that can be shared by all agent components.

However, since this computation (i.e. merging tags from config) is only done at the initialisation phase of other components (for example when initialising the tagger), it should be fine to do the computation each time. 

The benefit is to avoid global states. 

This is acceptable because:
- The computation is not heavy
- The computation is not done often